src/libraries/System.Private.CoreLib/src/Resources/Strings.resx

```diff
@@ -3410,6 +3410,9 @@
   <data name="SemaphoreSlim_Wait_TimeoutWrong" xml:space="preserve">
     <value>The timeout must represent a value between -1 and Int32.MaxValue, inclusive.</value>
   </data>
+  <data name="SemaphoreSlim_Wait_TimeSpanTimeoutWrong" xml:space="preserve">
+    <value>The timeout must represent a value between -1 and Timer.MaxSupportedTimeout, inclusive.</value>
```

Comment:
`MaxSupportedTimeout` is private implementation detail. It should not be used in user facing error messages.